### ISSUE TITLE
SSR for `sizes` is not producing the same results as unoptimized version or version from amp-toolbox

### ISSUE THREAD

**Issue by kswro** [ampproject/amp-toolbox-php#333]
## Bug Description

See test case in https://github.com/ampproject/amp-toolbox-php/issues/333#issuecomment-907393025.

----

I'm having a problem with AMP-WP running in my installation. In both standard and transitional mode (with ?amp=1), it seems that AMP-WP remove the `sizes` attribute in the `<amp-img>` tag of the template. The WordPress theme that I created completely uses the AMPHTML format.

I've also read the issue #https://github.com/ampproject/amp-wp/issues/4606, but what I'm experiencing right now is in the `<amp-img>` markup I wrote directly in the template, not from the time of conversion from `<img>` to `<amp-img>`.

One example can be found in the  markup (directly written in template, not from `the_content()` function) I use as follows:

```html
<amp-img
    data-hero 
    alt="Some Alt"
    srcset="http://xxxx/some-img-240x135.webp 240w, http://xxxx/some-img-480x270.webp 480w, http://xxxx/some-img-720x405.webp 720w, http://xxxx/some-img-960x540.webp 960w, http://xxxx/some-img-1200x675.webp 1200w"
    sizes="(max-width: 768px) 100vw, (max-width: 1023px) 64vw, (max-width: 1215px) 454px, 550px"
    src="http://xxxx/some-img-1200x675.webp"
    width="16"
    height="9"
    layout="responsive">
</amp-img>
```

For information, the value of the `srcset` attribute above is the output of the `wp_get_attachment_image_srcset()`, while the values ​​of the `sizes` attribute are entered directly into the template. Check this out: https://developer.wordpress.org/apis/handbook/responsive-images/#customizing-responsive-image-markup

Once AMP-WP is enabled, the markup changes to:

```html
<amp-img
    data-hero
    alt="Some Alt"
    srcset="http://xxxx/some-img-240x135.webp 240w, http://xxxx/some-img-480x270.webp 480w, http://xxxx/some-img-720x405.webp 720w, http://xxxx/some-img-960x540.webp 960w, http://xxxx/some-img-1200x675.webp 1200w"
    src="http://xxxx/some-img-1200x675.webp"
    width="16"
    height="9"
    layout="responsive"
    i-amphtml-ssr
    id="i-amp-2"
    class="i-amphtml-layout-responsive i-amphtml-layout-size-defined"
    i-amphtml-layout="responsive">
</amp-img>
```

Note that the `sizes` attribute disappears from the markup above and makes the layout of the image not look as expected.  On some other amp-imgs, I've also come across cases where the sizes values ​​in the template are changing from what they should be.

In another case, the `link rel="preload"` line with the `imgsizes` attribute as below, which I placed in the `<head>`, is also missing when the AMP-WP plugin is activated.

```html
<link rel="preload" as="image" imagesrcset="http://xxxx/img/logo-80.png 80w, http://xxxx/img/logo-107.png 107w, http://xxxx/img/logo-240.png 240w, http://xxxx/img/logo-480.png 480w" imagesizes="(max-width: 768px) 80px, 107px">
```

## Expected Behaviour

The plugin does not delete or change the values of the sizes attribute that has been set in the template 

As additional information, I have also tried to optimize with AMP Optimizer (node.js) on existing AMPHTML templates and did not encounter this problem.

Is this an error caused by AMP-WP or an error from my own side? 

Thank you in advance.

## Additional context

 - WordPress version: 5.8
 - Plugin version: 2.1.3
 - AMP plugin template mode: Standard and Transitional (with ?amp=1)
 - PHP version: 8.0.9
 - OS: Manjaro Linux 
 - Browser: Latest Chrome and Firefox

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

**Comment by kswro** [comment]
> @kswro So please confirm. Is thera problem with PHP (including the AMP-WP plugin) or is it an issue with the Node.js AMP Optimizer?

I can confirm that the problem lies with AMP Toolbox PHP.

![compare](https://user-images.githubusercontent.com/37024737/131203475-9581612d-e41d-4a96-97b4-9bcc2ed49924.jpg)

**Comment by kswro** [comment]
**Update:**
The striking difference that causes this problem I think lies in the output of media queries generated by the PHP Toolbox Optimizer.

If I copy the order of media queries from the output of Node.js to the optimization results from PHP, then the display becomes as desired.

![compare2](https://user-images.githubusercontent.com/37024737/131204202-c113e47f-5ef0-4dda-b476-ee99f1e6e542.png)

**Comment by schlessera** [comment]
I see what the problem is. The logic to merge multiple selectors that use the same rules does not consider the media query as part of the rules. I'll produce a fix.

**Comment by schlessera** [comment]
Hmm, it's actually more complicated than that. The ordering of multiple media queries plays a part here. I'm afraid the only solution at this point is to disable the CSS rule merging, as it will be too complex to solve this in a robust way.

**Comment by kswro** [comment]
For me the `disable-inline-width` attribute was enough to solve this problem. It's just that when I first experienced this I was confused why the amphtml display was different from what was produced by AMP-WP/Toolbox PHP which at first I thought the error was in my CSS code.

**Comment by schlessera** [comment]
@kswro Fix PR: #335

**Comment by kswro** [comment]
Yup, fixed now. Thank you... 

![fixed](https://user-images.githubusercontent.com/37024737/131211215-b7fb43aa-f2d3-4f0c-8e03-9b75ad41e4db.png)
